subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,588 @@
+
+[[migmvn:common_plugins]]
+== Migrating common plugins
+
+Maven and Gradle share a common approach of extending the build through plugins.
+Although the plugin systems are very different beneath the surface, they share many feature-based plugins, such as:
+
+* Shade/Shadow
+* Jetty
+* Checkstyle
+* JaCoCo
+* AntRun (see further down)
+
+Why does this matter? Because many plugins rely on standard Java conventions, so migration is just a matter of replicating the configuration of the Maven plugin in Gradle.
+As an example, here's a simple Maven Checkstyle plugin configuration:
+
+[source,xml]
+----
+...
+<plugin>
+  <groupId>org.apache.maven.plugins</groupId>
+  <artifactId>maven-checkstyle-plugin</artifactId>
+  <version>2.17</version>
+  <executions>
+    <execution>
+      <id>validate</id>
+      <phase>validate</phase>
+      <configuration>
+        <configLocation>checkstyle.xml</configLocation>
+        <encoding>UTF-8</encoding>
+        <consoleOutput>true</consoleOutput>
+        <failsOnError>true</failsOnError>
+        <linkXRef>false</linkXRef>
+      </configuration>
+      <goals>
+        <goal>check</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+...
+----
+
+Everything outside of the configuration block can safely be ignored when migrating to Gradle.
+In this case, the corresponding Gradle configuration looks like the following:
+
+.Configuring the Gradle Checkstyle Plugin
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=checkstyle]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=checkstyle]"]
+====
+
+The Checkstyle tasks are automatically added as dependencies of the `check` task, which also includes `test`.
+If you want to ensure that Checkstyle runs before the tests, then just specify an ordering with the mustRunAfter() method:
+
+.Controlling when the `checkstyle` task runs
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=depends-on]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=depends-on]"]
+====
+
+As you can see, the Gradle configuration is often much shorter than the Maven equivalent.
+You also have a much more flexible execution model since you are no longer constrained by Maven's fixed phases.
+
+While migrating a project from Maven, don't forget about source sets.
+These often provide a more elegant solution for handling integration tests or generated sources than Maven can provide, so you should factor them into your migration plans.
+
+
+=== Ant goals
+
+Many Maven builds rely on the AntRun plugin to customize the build without the overhead of implementing a custom Maven plugin.
+Gradle has no equivalent plugin because Ant is a first-class citizen in Gradle builds, via the `ant` object.
```

Comment:
but I would recommend to create custom tasks in `buildSrc` instead of delegating to Ant, whenever possible.